HELSSA_AGENTS/admin_dashboard/app_code/cores/text_processor.py

```
@@ -0,0 +1,366 @@
+            result = self._structure_ai_response(
+                original_text=text,
+                ai_response=ai_response,
+                processing_time=time.time() - start_time
+            )
+            
+            self.logger.info(f"Medical text processing completed in {result.processing_time:.2f}s")
+            return result
+            
+        except Exception as e:
+            self.logger.error(f"Medical text processing error: {str(e)}")
+            raise Exception(f"خطا در پردازش متن پزشکی: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)